contracts/other/notional/NotionalMultiOracle.sol

```diff
@@ -0,0 +1,81 @@
+import "@yield-protocol/utils-v2/contracts/cast/CastBytes32Bytes6.sol";
+import "@yield-protocol/utils-v2/contracts/token/IERC20Metadata.sol";
+import "@yield-protocol/vault-interfaces/IOracle.sol";
+
+
+/**
+ * @title NotionalMultiOracle
+ * @notice We value fCash assets at face value
+ */
+contract NotionalMultiOracle is IOracle, AccessControl {
+    using CastBytes32Bytes6 for bytes32;
+
+    event SourceSet(bytes6 indexed notionalId, bytes6 indexed underlyingId, address underlying);
+
+    struct Source {
+        uint8 baseDecimals;
+        uint8 quoteDecimals;
+        bool inverse;
+    }
+
+    mapping(bytes6 => mapping(bytes6 => Source)) public sources;
+
+    /// @dev Set or reset an oracle source and its inverse
+    function setSource(bytes6 notionalId, bytes6 underlyingId, IERC20Metadata underlying)
+        external auth
+    {
+        require (notionalId != underlyingId, "Wrong input");
+        sources[notionalId][underlyingId] = Source({
+            baseDecimals: 18, // I'm assuming here that fCash has 18 decimals
+            quoteDecimals: underlying.decimals(), // Ideally we would get the underlying from fCash
+            inverse: false
+        });
+        emit SourceSet(notionalId, underlyingId, address(underlying));
+
+        sources[underlyingId][notionalId] = Source({
+            baseDecimals: underlying.decimals(), // We are reversing the base and the quote
+            quoteDecimals: 18,
+            inverse: true
+        });
+        emit SourceSet(underlyingId, notionalId, address(underlying));
+    }
+
+    /// @dev Convert amountBase base into quote at the latest oracle price.
+    function peek(bytes32 baseId, bytes32 quoteId, uint256 amountBase)
+        external view virtual override
+        returns (uint256 amountQuote, uint256 updateTime)
+    {
+        if (baseId == quoteId) return (amountBase, block.timestamp);
+        (amountQuote, updateTime) = _peek(baseId.b6(), quoteId.b6(), amountBase);
+    }
+
+    /// @dev Convert amountBase base into quote at the latest oracle price, updating state if necessary. Same as `peek` for this oracle.
+    function get(bytes32 baseId, bytes32 quoteId, uint256 amountBase)
+        external virtual override
+        returns (uint256 amountQuote, uint256 updateTime)
+    {
+        if (baseId == quoteId) return (amountBase, block.timestamp);
+        (amountQuote, updateTime) = _peek(baseId.b6(), quoteId.b6(), amountBase);
+    }
+
+    /// @dev Convert amountBase base into quote at the latest oracle price.
+    function _peek(bytes6 baseId, bytes6 quoteId, uint256 amountBase)
+        private view
+        returns (uint amountQuote, uint updateTime)
+    {
+        Source memory source = sources[baseId][quoteId];
+        require (source.baseDecimals == 18 || source.quoteDecimals == 18, "Source not found"); // A bit meh as a test of existence
+        int price = 1e18; // We price fCash at face value
```

Comment:
i don't know much about fcash but i see we are always valuing it at 1:1

is all this really necessary?  Can't we just `return (amount, block.timestamp)` always and call it a day?

Reply:
We can value fCash at 1:1 because we are still using a collateralization ratio/loan-to-value. I believe the loan-to-value is 90%, meaning that the position is still above water even if interest rates hit 20% for a six-month term. Even if interest rates go above that, Yield Protocol doesn't lose money as it could always redeem the fCash for the underlying asset itself, even if the user never repays the loan.

Reply:
I see.  What is different about fcash that we can use 1, but Dai or USDC we have to check prices?  We also use a collateralization ratio for those.

Reply:
fCash Dai settles to Dai at 1:1. So, we can value fCash Dai as being equal to Dai, with the only risk being that if we are wrong at that time, we may have to wait until maturity for it to be true. This assumption only works well if the collateral has the same (or an earlier) maturity than what is borrowed.

Reply:
@alcueca I understand better the thinking behinding valuing these at 1, but i still don't understand why we need all these steps if it's always going to return amount in the end.  It's a waste of gas and confusing. 

> Can't we just return (amount, block.timestamp) always and call it a day?

Reply:
Because fCash has 8 decimals, the base could have 18 (DAI) or 6 (USDC), so we need to convert from 1000000 USDC to 100000000 fUSDC, or 100000000 fDAI to 1000000000000000000 DAI.

Reply:
Simplified the code a bit, tho.